### ISSUE TITLE
Confused about arrays

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1272]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!bash

edit='emacsclient -c'
declare -a editcmd
if [ -n "$edit" ]
then
    read -r -a editcmd < <(echo "$edit")
fi

if [[ "$editcmd" =~ emacs\  ]]
then
    editcmd="$editcmd -g +100+100"
fi

if [ -n "$editcmd" ]
then
    "$editcmd" &
fi

```

#### Here's what shellcheck currently says:
```

```
#### Here's what I wanted or expected to see:

I expected it to complain about using "$editcmd" instead of "${editcmd[@]}" in all the places where the former was used. Am I missing something?

**Comment by kerframil** [comment]
I think that this ought to trigger SC2128. Consider the following sample.

```sh
#!/bin/bash
var=(foo bar baz)
[[ $var == foo\ * ]]
```

Shellcheck reports on it as follows:-

```
[[ $var == foo\ * ]]
   ^-- SC2128: Expanding an array without an index only gives the first element.
```

The use of read -ra seems to trip shellcheck up.

**Comment by kerframil** [comment]
See also #679.

**Comment by koalaman** [comment]
Fixed by @ngzhian. Thanks!
